lib/engines/content_block_manager/test/unit/app/services/get_preview_content_test.rb

```diff
@@ -4,29 +4,44 @@ class ContentBlockManager::GetPreviewContentTest < ActiveSupport::TestCase
   extend Minitest::Spec::DSL
 
   let(:described_class) { ContentBlockManager::GetPreviewContent }
-  let(:host_content_id) { "64570503-7a7f-4fca-80c1-e6dce7278419" }
+  let(:host_content_id) { SecureRandom.uuid }
+  let(:preview_content_id) { SecureRandom.uuid }
   let(:host_title) { "Test" }
   let(:host_base_path) { "/test" }
   let(:uri_mock) { mock }
   let(:fake_frontend_response) do
-    "<!DOCTYPE html><body><p>test</p></body>"
+    "<body><p>test</p><span class=\"content-embed content-embed__content_block_email_address\" data-content-block=\"\" data-document-type=\"content_block_email_address\" data-content-id=\"#{preview_content_id}\">example@example.com</span></body>"
+  end
+  let(:expected_html) do
+    "<body><p>test</p><span class=\"content-embed content-embed__content_block_email_address\" data-content-block=\"\" data-document-type=\"content_block_email_address\" data-content-id=\"#{preview_content_id}\">new@new.com</span></body>"
+  end
+  let(:document) do
+    build(:content_block_document, :email_address, content_id: preview_content_id)
+  end
+  let(:block_to_preview) do
+    build(:content_block_edition, :email_address, document:, details: { "email_address" => "new@new.com" })
   end
 
   describe "#preview_content" do
     setup do
       stub_publishing_api_has_item(content_id: host_content_id, title: host_title, base_path: host_base_path)
     end
 
-    it "returns the title and raw frontend HTML for a document" do
+    it "returns the title and preview HTML for a document" do
       Net::HTTP.expects(:get).with(URI(Plek.website_root + host_base_path)).returns(fake_frontend_response)
```

Comment:
we are no longer strictly unit testing the Nokogiri functionality, it seemed way to complicated to stub its methods as we iterate through the different elements, and I think it's worth letting that slide to get the assurance we're getting the content we expect?

Reply:
Yeah, that makes sense I think